tests/api2/test_smb_client.py

```diff
@@ -0,0 +1,185 @@
+import os
+import pytest
+
+from middlewared.test.integration.assets.account import user, group
+from middlewared.test.integration.assets.pool import dataset
+from middlewared.test.integration.assets.smb import smb_share, smb_mount
+from middlewared.test.integration.utils import call, client, ssh
+
+
+PERMSET = {
+    "READ_DATA": False,
+    "WRITE_DATA": False,
+    "APPEND_DATA": False,
+    "READ_NAMED_ATTRS": False,
+    "WRITE_NAMED_ATTRS": False,
+    "EXECUTE": False,
+    "DELETE_CHILD": False,
+    "READ_ATTRIBUTES": False,
+    "WRITE_ATTRIBUTES": False,
+    "DELETE": False,
+    "READ_ACL": False,
+    "WRITE_ACL": False,
+    "WRITE_OWNER": False,
+    "SYNCHRONIZE": True
+}
+
+SAMPLE_ENTRY = {
+    "tag": "GROUP",
+    "id": 666,
+    "type": "ALLOW",
+    "perms": PERMSET,
+    "flags": {"BASIC": "INHERIT"}
+}
+
+PERSISTENT_ACL = [
+    {
+        "tag": "GROUP",
+        "id": 545,
+        "type": "ALLOW",
+        "perms": {"BASIC": "FULL_CONTROL"},
+        "flags": {"BASIC": "INHERIT"}
+    }
+]
+
+@pytest.fixture(scope='module')
+def setup_smb_tests(request):
+    with dataset('smbclient-testing', data={'share_type': 'SMB'}) as ds:
+        with user({
+            'username': 'smbuser',
+            'full_name': 'smbuser',
+            'group_create': True,
+            'password': 'Abcd1234$'
+        }) as u:
+            with smb_share(os.path.join('/mnt', ds), 'client_share') as s:
+                try:
+                    call('service.start', 'cifs')
+                    yield {'dataset': ds, 'share': s, 'user': u}
+                finally:
+                    call('service.stop', 'cifs')
+
+
+@pytest.fixture(scope='module')
+def mount_share(setup_smb_tests):
+    with smb_mount(setup_smb_tests['share']['name'], 'smbuser', 'Abcd1234$') as mp:
```

Comment:
I don't see `Abcd1234$` constant defined anywhere else in the project, should we make it a constant?